### PR TITLE
Cleaned Up Excel Commands

### PR DESCRIPTION
**Pull request by openbots-ff**

### CODE REVIEW

File: taskt/Core/Automation/Commands/Excel Commands/ExcelAppendCellCommand.cs [hunk 1]
```diff
@@ -1,80 +1,78 @@
...
-            int test = 0;
-            test = excelSheet.Columns.Count;
+            var excelInstance = (Application)excelObject;
+            Worksheet excelSheet = excelInstance.ActiveSheet;
+  
             var lastUsedRow = excelSheet.Cells.Find("*", System.Reflection.Missing.Value,
                                  System.Reflection.Missing.Value, System.Reflection.Missing.Value,
-                                 Microsoft.Office.Interop.Excel.XlSearchOrder.xlByRows, Microsoft.Office.Interop.Excel.XlSearchDirection.xlPrevious,
+                                 XlSearchOrder.xlByRows, XlSearchDirection.xlPrevious,
                                  false, System.Reflection.Missing.Value, System.Reflection.Missing.Value).Row;
```
Comment (ob-ab) [inline taskt/Core/Automation/Commands/Excel Commands/ExcelAppendCellCommand.cs]: Please clean up `System.Reflection` namespace

File: taskt/Core/Automation/Commands/Excel Commands/ExcelCreateApplicationCommand.cs [hunk 2]
```diff
@@ -2,67 +2,73 @@
...
         [XmlAttribute]
-        [Attributes.PropertyAttributes.PropertyDescription("Please Enter the instance name")]
-        [Attributes.PropertyAttributes.InputSpecification("Signifies a unique name that will represemt the application instance.  This unique name allows you to refer to the instance by name in future commands, ensuring that the commands you specify run against the correct application.")]
-        [Attributes.PropertyAttributes.SampleUsage("**myInstance** or **excelInstance**")]
-        [Attributes.PropertyAttributes.Remarks("")]
-        [Attributes.PropertyAttributes.PropertyUIHelper(Attributes.PropertyAttributes.PropertyUIHelper.UIAdditionalHelperType.ShowVariableHelper)]
+        [PropertyDescription("Excel Instance Name")]
+        [InputSpecification("Signifies a unique name that will represent the application instance. " +
+            "This unique name allows you to refer to the instance by name in future commands, " +
+            "ensuring that the commands you specify run against the correct application.")]
```
Comment (ob-ab) [inline taskt/Core/Automation/Commands/Excel Commands/ExcelCreateApplicationCommand.cs]: You can simply keep the input specification short like "Enter a unique name that will represent the excel application instance."

You can put the extra information in Remarks. Also remove the "allows you to" part from the text.

File: taskt/Core/Automation/Commands/Excel Commands/ExcelDeleteRangeCommand.cs [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+using taskt.Core.Utilities.CommonUtilities;
+using taskt.UI.CustomControls;
+using taskt.UI.Forms;
+using Application = Microsoft.Office.Interop.Excel.Application;
+
+namespace taskt.Core.Automation.Commands
+{
+    [Serializable]
+    [Group("Excel Commands")]
+    [Description("This command deletes a specific cell in an Excel Worksheet.")]
```
Comment (ob-ab) [inline taskt/Core/Automation/Commands/Excel Commands/ExcelDeleteRangeCommand.cs]: Please update the description as this command deletes a range not only a specific cell.

File: taskt/Core/Automation/Commands/Excel Commands/ExcelCreateApplicationCommand.cs (hunk 2 above)
Comment (openbots-ff) [reply taskt/Core/Automation/Commands/Excel Commands/ExcelCreateApplicationCommand.cs]: Okay, I'll split this up, but I left you a similar comment in your IE browser commands so please check that they match

Comment (openbots-ff) [reply taskt/Core/Automation/Commands/Excel Commands/ExcelCreateApplicationCommand.cs]: I think this particular use of allows is necessary though

File: taskt/Core/Automation/Commands/Excel Commands/ExcelSaveWorkbookAsCommand.cs [hunk 4]
```diff
@@ -1,79 +1,86 @@
...
+
         [XmlAttribute]
-        [Attributes.PropertyAttributes.PropertyDescription("Please indicate the directory of the file")]
-        [Attributes.PropertyAttributes.PropertyUIHelper(Attributes.PropertyAttributes.PropertyUIHelper.UIAdditionalHelperType.ShowVariableHelper)]
-        [Attributes.PropertyAttributes.PropertyUIHelper(Attributes.PropertyAttributes.PropertyUIHelper.UIAdditionalHelperType.ShowFileSelectionHelper)]
-        [Attributes.PropertyAttributes.InputSpecification("Enter or Select the path to the file.")]
-        [Attributes.PropertyAttributes.SampleUsage("C:\\temp\\myfile.xlsx or [vExcelFilePath]")]
-        [Attributes.PropertyAttributes.Remarks("")]
+        [PropertyDescription("Workbook Location")]
+        [InputSpecification("Enter or Select the path of the folder to save the Workbook to.")]
```
Comment (Ali-Basharat) [inline taskt/Core/Automation/Commands/Excel Commands/ExcelSaveWorkbookAsCommand.cs:32]: You can also add the ShowFolderSelectionHelper for this property.

File: taskt/Core/Automation/Commands/Excel Commands/ExcelAppendCellCommand.cs (hunk 1 above)
Comment (openbots-ff) [reply taskt/Core/Automation/Commands/Excel Commands/ExcelAppendCellCommand.cs]: done

File: taskt/Core/Automation/Commands/Excel Commands/ExcelDeleteRangeCommand.cs (hunk 3 above)
Comment (openbots-ff) [reply taskt/Core/Automation/Commands/Excel Commands/ExcelDeleteRangeCommand.cs]: done

File: taskt/Core/Automation/Commands/Excel Commands/ExcelWriteCellCommand.cs [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        [PropertyDescription("Cell Value")]
+        [InputSpecification("Enter the text value that will be set in the selected cell.")]
+        [SampleUsage("Hello World || {vText}")]
+        [Remarks("")]
+        [PropertyUIHelper(PropertyUIHelper.UIAdditionalHelperType.ShowVariableHelper)]
+        public string v_TextToSet { get; set; }
+
+        [XmlAttribute]
+        [PropertyDescription("Cell Location")]
+        [InputSpecification("Enter the location of the cell to extract.")]
```
Comment (ob-ab) [inline taskt/Core/Automation/Commands/Excel Commands/ExcelWriteCellCommand.cs]: Change to "Enter the location of the cell to set the text value."

Comment (openbots-ff) [reply taskt/Core/Automation/Commands/Excel Commands/ExcelWriteCellCommand.cs]: done

**Review by ob-ab — CHANGES_REQUESTED** [review]
Please make a few minor changes.

**Review by ob-ab — COMMENTED** [review]
Looks good.

File: taskt/Core/Automation/Commands/Excel Commands/ExcelActivateCellCommand.cs [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    [Group("Excel Commands")]
+    [Description("This command activates a specific cell in an Excel Worksheet.")]
+
+    public class ExcelActivateCellCommand : ScriptCommand
+    {
+        [XmlAttribute]
+        [PropertyDescription("Excel Instance Name")]
+        [InputSpecification("Enter the unique instance that was specified in the **Create Application** command.")]
+        [SampleUsage("MyExcelInstance || {vExcelInstance}")]
+        [Remarks("Failure to enter the correct instance or failure to first call **Create Application** command will cause an error.")]
```
Comment (CameronHerwig) [inline taskt/Core/Automation/Commands/Excel Commands/ExcelActivateCellCommand.cs]: Change to "first call the".

File: taskt/Core/Automation/Commands/Excel Commands/ExcelCreateApplicationCommand.cs [hunk 7]
```diff
@@ -2,67 +2,72 @@
...
+        [SampleUsage("MyExcelInstance")]
+        [Remarks("This unique name allows you to refer to the instance by name in future commands, " +
+                 "ensuring that the commands you specify run against the correct application.")]
         public string v_InstanceName { get; set; }
 
+        [XmlAttribute]
+        [PropertyDescription("Visible")]
+        [PropertyUISelectionOption("Yes")]
+        [PropertyUISelectionOption("No")]
+        [InputSpecification("Indicate whether Excel automation should be visible or not.")]
```
Comment (CameronHerwig) [inline taskt/Core/Automation/Commands/Excel Commands/ExcelCreateApplicationCommand.cs]: Change to "whether the Excel".

File: taskt/Core/Automation/Commands/Excel Commands/ExcelGetLastRowIndexCommand.cs [hunk 8]
```diff
@@ -1,84 +1,85 @@
...
 
 namespace taskt.Core.Automation.Commands
 {
     [Serializable]
-    [Attributes.ClassAttributes.Group("Excel Commands")]
-    [Attributes.ClassAttributes.Description("This command allows you to find the last row in a used range in an Excel Workbook.")]
-    [Attributes.ClassAttributes.UsesDescription("Use this command to determine how many rows have been used in the Excel Workbook.  You can use this value in a **Number Of Times** Loop to get data.")]
-    [Attributes.ClassAttributes.ImplementationDescription("This command implements Excel Interop to achieve automation.")]
+    [Group("Excel Commands")]
+    [Description("This command gets the last row index of a range in an Excel Worksheet.")]
```
Comment (CameronHerwig) [inline taskt/Core/Automation/Commands/Excel Commands/ExcelGetLastRowIndexCommand.cs]: Change to "This commands retrieves the index of the last row of a range"

**Review by CameronHerwig — CHANGES_REQUESTED** [review]

File: taskt/Core/Automation/Commands/Excel Commands/ExcelActivateCellCommand.cs (hunk 6 above)
Comment (CameronHerwig) [reply taskt/Core/Automation/Commands/Excel Commands/ExcelActivateCellCommand.cs]: This needs repeated in most files. I would recommend find and replace.

Comment (openbots-ff) [reply taskt/Core/Automation/Commands/Excel Commands/ExcelActivateCellCommand.cs]: done

File: taskt/Core/Automation/Commands/Excel Commands/ExcelCreateApplicationCommand.cs (hunk 7 above)
Comment (openbots-ff) [reply taskt/Core/Automation/Commands/Excel Commands/ExcelCreateApplicationCommand.cs]: done

File: taskt/Core/Automation/Commands/Excel Commands/ExcelGetLastRowIndexCommand.cs (hunk 8 above)
Comment (openbots-ff) [reply taskt/Core/Automation/Commands/Excel Commands/ExcelGetLastRowIndexCommand.cs]: done

**Review by CameronHerwig — DISMISSED** [review]

File: taskt/Core/Automation/Commands/Excel Commands/ExcelAppendRowCommand.cs [hunk 9]
```diff
@@ -1,98 +1,160 @@
...
+            RenderedControls.AddRange(CommandControls.CreateDefaultInputGroupFor("v_RowToSet", this, editor));
 
             return RenderedControls;
-
         }
+
         public override string GetDisplayValue()
         {
-            return base.GetDisplayValue() + " [Append Row '" +v_TextToSet+ " to last row of workboook with Instance Name: '" + v_InstanceName + "']";
+            return base.GetDisplayValue() + $" [Append '{v_RowToSet}' - Instance Name: '{v_InstanceName}']";
```
Comment (ob-ab) [inline taskt/Core/Automation/Commands/Excel Commands/ExcelAppendRowCommand.cs]: Please remove colon

**Review by ob-ab — CHANGES_REQUESTED** [review]
Please make a minor change.

File: taskt/Core/Automation/Commands/Excel Commands/ExcelAppendRowCommand.cs (hunk 9 above)
Comment (openbots-ff) [inline taskt/Core/Automation/Commands/Excel Commands/ExcelAppendRowCommand.cs]:
```suggestion
            return base.GetDisplayValue() + $" [Append '{v_RowToSet}' - Instance Name '{v_InstanceName}']";
```

Comment (openbots-ff) [reply taskt/Core/Automation/Commands/Excel Commands/ExcelAppendRowCommand.cs]: done

**Review by CameronHerwig — APPROVED** [review]

**Review by ob-ab — APPROVED** [review]

**Pull request merged**